test/integration/nighthawk_grpc_service.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import socket
+import subprocess
+import tempfile
+import threading
+import time
+
+from common import IpVersion
+
+
+# TODO(oschaaf): unify some of this code with the test server wrapper.
+class NighthawkGrpcService(object):
+  """
+  Class for running the Nighthawk gRPC service in a separate process.
+  Usage:
+    grpc_service = NighthawkGrpcService("/path/to/nighthawk_service"), "127.0.0.1", IpVersion.IPV4)
+    if grpc_service.start():
+      .... You can talk to the Nighthawk gRPC service at the 127.0.0.1:grpc_service.server_port ...
+
+  Attributes:
+  server_ip: IP address used by the gRPC service to listen. 
+  server_port: An integer, indicates the port used by the gRPC service to listen. 0 means that the server is not listening.
+  """
+
+  def __init__(self, server_binary_path, server_ip, ip_version):
+    """Initializes Nighthawk gRPC service.
+
+    Args:
+    server_binary_path: A string, indicates where the nighthawk gRPC service binary resides
+    server_ip: IP address, indicates which ip address should be used by the gRPC service listener.
+    ip_version: IP Version, indicates if IPv4 or IPv6 should be used.
+    ...
+    """
+    assert ip_version != IpVersion.UNKNOWN
+    self.server_port = 0
+    self.server_ip = server_ip
+    self._server_process = None
+    self._ip_version = ip_version
+    self._server_binary_path = server_binary_path
+    self._socket_type = socket.AF_INET6 if ip_version == IpVersion.IPV6 else socket.AF_INET
+    self._server_thread = threading.Thread(target=self._serverThreadRunner)
+    self._address_file = ""
+
+  def _serverThreadRunner(self):
+    with tempfile.NamedTemporaryFile(mode="w", delete=False, suffix=".tmp") as tmp:
```

Comment:
Isn't the benefit of having a context manager here that it will delete/clean up for us when we're done? Why not run the code below under it?